pkg/blockbuilder/kafkautil.go

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package blockbuilder
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kerr"
+)
+
+// getGroupLag is the inlined version of `kadm.Client.Lag` but updated to work with when the group
+// doesn't have live participants.
+// Similar to `kadm.CalculateGroupLagWithStartOffsets`, it takes into account that the group may not have any commits.
+func getGroupLag(ctx context.Context, admClient *kadm.Client, topic, group string, fallbackOffset int64) (kadm.GroupLag, error) {
```

Comment:
We briefly talked about this `getGroupLag` chunk @dimitarvdimitrov, asking if we should aim at keeping this utils in the forked version of `kgo`.

Thinking a bit more, I don't think I'm fancy about that idea. That is, in my mind, these are simple helper functions, that do "maths" against four sets of offsets (partition's start, end, commit, fallback). It's not something that we're "fixing" in kgo.

Instead, moving forward, I'd rather simplified this code, with the idea that our consumers never consume more than one topic/partition. I.e. we don't even need the complexity of `kadm.GroupLag` struct here.